server/services/curiousService.ts

```diff
@@ -0,0 +1,122 @@
+import { startOfDay, subMonths } from 'date-fns'
+import CuriousApiClient from '../data/interfaces/curiousApi/curiousApiClient'
+import logger from '../../logger'
+import PrisonService from './prisonService'
+import { InPrisonCourseRecords, InPrisonCourse } from './interfaces/curiousService/CuriousInPrisonCourses'
+import { LearnerEducation, LearnerEductionPagedResponse } from '../data/interfaces/curiousApi/LearnerEducation'
+import { toInPrisonCourse } from './mappers/inPrisonCourseMapper'
+import dateComparator from '../utils/dateComparator'
+import { RestClientBuilder } from '../data'
+
+export default class CuriousService {
```

Comment:
`CuriousService` is a new class, it didnt exist before. Previously the classes that need to get Curious Course & Qualification data do so by using the REST API client and calling the REST API directly.

That approach might be OK (?) based on the current implementation where it simply calls the API and maps the response into a simplified data model for the view; but now we have some logic to content with in respect of calling the service until we get the last page. A service class is the right abstraction for this 👍 

At the moment this service class only has the method for getting the Curious courses & qualifications because thats all we are concerned with in this scope of work.
But over time the other Curious API calls to `/learnerProfile` etc should be moved here